doc/guides/west/build-flash-debug.rst

```diff
@@ -27,88 +27,175 @@ directly delegate to West.
 Building: ``west build``
 ************************
 
-.. tip:: Run ``west build -h`` for additional help.
+.. tip:: Run ``west build -h`` for a quick overview.
 
-The ``build`` command allows you to build any source tree from any directory
-in your file system, placing the build results in a folder of your choice.
+The ``build`` command helps you build Zephyr applications from source. You can
+use :ref:`west config <west-config-cmd>` to configure its behavior.
 
-In its simplest form, the command can be run by navigating to the root folder
-(i.e. the folder containing a :file:`CMakeLists.txt` file) of the Zephyr
-application of your choice and running::
+This command attempts to "do what you mean" when run from a Zephyr application
+source or build directory:
+
+- When running ``west build`` in an existing build directory, the board, source
```

Comment:
Better grammar would be:
```
When you run ``west build`` ...
```
It's like the sentence, "While walking down the street, the trees were waving in the wind."  where it's as if the trees were walking down the street.